Урок 8. Корреляционный анализ
1) Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy

Полученные значения должны быть равны.

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

2) Измерены значения IQ выборки студентов,

обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95.

3) Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

1) Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy

Полученные значения должны быть равны.

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [2]:
import numpy as np
import scipy.stats as st
import pandas as pd

In [3]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

cov = np.mean(zp*ks) - np.mean(zp) * np.mean(ks)
print(cov)

9157.839999999997


np.cov(zp, ks, ddof = 0)

In [24]:
#Найдем коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков
print(np.std(zp, ddof = 0))
print(np.std(ks, ddof = 0))

res = cov / (np.std(zp, ddof = 0)*np.std(ks, ddof = 0))
res

59.115480206118605
174.55340157098058


0.8874900920739158

In [20]:
#Найдем коэффициент корреляции Пирсона с помощью numpy
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [25]:
#Найдем коэффициент корреляции Пирсона с помощью numpy
df=pd.DataFrame(data={'zp':zp, 'ks':ks})
df
df.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

In [ ]:
2) Измерены значения IQ выборки студентов,

обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [29]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

mean = np.mean(iq)
N = len(iq)

print(mean)
print(N)

118.1
10


In [32]:
std = np.std(iq, ddof=1)

In [31]:
alpha = 0.05 
df = len(iq) - 1

#Табличное значение t-критерия
t = st.t.ppf(1 - alpha / 2, df)
t

2.2621571627409915

In [33]:
lower = mean - t * (std / N ** 0.5)
upper = mean + t * (std / N ** 0.5)

print(lower, upper)

110.55608365158724 125.64391634841274


Ответ: Доверительный интервал равен:
(110.55608365158724, 125.64391634841274)

3) Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [34]:
#Z-критерий
lower = 174.2 - (1.96 * 25**0.5) / 27**0.5
upper = 174.2 + (1.96 * 25**0.5) / 27**0.5
print(lower, upper)

172.31398912064722 176.08601087935276


Ответ: Доверительный интервал равен:
(172.31398912064722 176.08601087935276)